finagle-base-http/src/main/scala/com/twitter/finagle/http/headers/JTreeMapBackedHeaderMap.scala

```diff
@@ -0,0 +1,79 @@
+package com.twitter.finagle.http.headers
+
+import com.twitter.finagle.http.HeaderMap
+import scala.annotation.tailrec
+
+/**
+ * Mutable, thread-safe [[HeaderMap]] implementation, backed by
+ * a mutable [[Map[String, Header]]], where the map key
+ * is forced to lower case
+ */
+final private[http] class JTreeMapBackedHeaderMap extends JMapBackedHeaderMap {
```

Comment:
What is the value of separating this from `JMapBackedHeaderMap`? I don't anticipate we'll want to change the underlying map all that often.

Reply:
fwiw, I'm not strongly opinionated about this, it's just that having an abstract backing map that is synchronized on in two different traits/classes makes me queasy.

Reply:
It made sense when the HashMap variety still existed. Not so much anymore.